addons/html_editor/static/src/main/align/align_plugin.js

```diff
@@ -0,0 +1,123 @@
+                id: "alignRight",
+                run: () => this.setAlignment("right"),
+            },
+            {
+                id: "justify",
+                run: () => this.setAlignment("justify"),
+            },
+        ],
+        toolbar_groups: withSequence(29, { id: "alignment" }),
+        toolbar_items: [
+            {
+                id: "alignment",
+                groupId: "alignment",
+                title: _t("Text align"),
+                Component: AlignSelector,
+                props: {
+                    getItems: () => alignmentItems,
+                    getDisplay: () => this.alignment,
+                    onSelected: (item) => {
+                        this.setAlignment(item.mode);
+                    },
+                },
+            },
+        ],
+
+        /** Handlers */
+        selectionchange_handlers: this.updateAlignmentParams.bind(this),
+        post_undo_handlers: this.updateAlignmentParams.bind(this),
+        post_redo_handlers: this.updateAlignmentParams.bind(this),
+        remove_format_handlers: this.setAlignment.bind(this),
+
+        /** Predicates */
+        has_format_predicates: hasBlockTextAlignment,
+    };
+
+    setup() {
+        this.alignment = reactive({ displayName: "" });
+    }
+
+    get alignmentMode() {
+        const sel = this.dependencies.selection.getSelectionData().deepEditableSelection;
+        const block = closestBlock(sel?.anchorNode);
+        const textAlign = this.getTextAlignment(block);
+        return ["center", "right", "justify"].includes(textAlign) ? textAlign : "left";
+    }
+
+    getTextAlignment(block) {
+        const { direction, textAlign } = getComputedStyle(block);
+        if (textAlign === "start") {
+            return direction === "rtl" ? "right" : "left";
+        } else if (textAlign === "end") {
+            return direction === "rtl" ? "left" : "right";
+        }
+        return textAlign;
+    }
+
+    setAlignment(mode = "") {
+        const visitedBlocks = new Set();
+        const traversedNode = this.dependencies.selection.getTraversedNodes();
+        let isAlignmentUpdated = false;
+
+        for (const node of traversedNode) {
+            if (isVisibleTextNode(node) || node.nodeName === "BR") {
+                const block = closestBlock(node);
+                if (!visitedBlocks.has(block)) {
+                    const currentTextAlign = this.getTextAlignment(block);
+                    if (currentTextAlign !== mode && block.isContentEditable) {
+                        block.style.textAlign = mode;
+                        isAlignmentUpdated = true;
+                    }
+                    visitedBlocks.add(block);
+                }
+            }
+        }
+        if (isAlignmentUpdated) {
+            this.dependencies.history.addStep();
+        }
+        this.updateAlignmentParams();
+    }
+
+    updateAlignmentParams() {
+        this.alignment.displayName = this.alignmentMode;
+    }
+}
```

Comment:
There are many things I don't like in this file but I noticed they are exactly the same in `font_plugin.js` since https://github.com/odoo/odoo/pull/183153 that I _did_ merge, so... maybe one day we'll want to refactor those but not today.